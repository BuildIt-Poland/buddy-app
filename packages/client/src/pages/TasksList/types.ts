import { RouteComponentProps } from 'react-router-dom';

export type TasksListProps = Pick<RouteComponentProps, 'history'>;
