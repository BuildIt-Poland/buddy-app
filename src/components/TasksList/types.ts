import { SnackBarProps } from 'decorators/withSnackBar';

export interface TaskListProps extends SnackBarProps {}
