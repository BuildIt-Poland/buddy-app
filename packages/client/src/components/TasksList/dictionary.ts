interface TaskListDictionary {
  BUDDY_TAB_TITLE: string;
  NEWBIE_TAB_TITLE: string;
  SUCCESS_MESSAGE: string;
  ERROR_MESSAGE: string;
  PLUS_BUTTON_TITLE: string;
}

const TASK_LIST_DICTIONARY: TaskListDictionary = {
  BUDDY_TAB_TITLE: 'My tasks',
  NEWBIE_TAB_TITLE: 'Newbie Tasks',
  SUCCESS_MESSAGE: 'Task status updated',
  ERROR_MESSAGE: 'An error ocurred updating task',
  PLUS_BUTTON_TITLE: 'Add new task',
};

export default TASK_LIST_DICTIONARY;
