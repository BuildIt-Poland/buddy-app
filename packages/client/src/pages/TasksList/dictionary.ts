interface TaskListDictionary {
  BUDDY_TAB_TITLE: string;
  NEWBIE_TAB_TITLE: string;
  SUCCESS_MESSAGE: string;
  ERROR_MESSAGE: string;
  TEMPLATE_MESSAGE: string;
  PLUS_BUTTON_TITLE: string;
}

const TASK_LIST_DICTIONARY: TaskListDictionary = {
  BUDDY_TAB_TITLE: 'My tasks',
  NEWBIE_TAB_TITLE: 'Newbie tasks',
  SUCCESS_MESSAGE: 'Task status updated',
  ERROR_MESSAGE: 'An error ocurred updating task',
  TEMPLATE_MESSAGE: 'Task status is static for template tasks',
  PLUS_BUTTON_TITLE: 'Add new task',
};

export default TASK_LIST_DICTIONARY;
