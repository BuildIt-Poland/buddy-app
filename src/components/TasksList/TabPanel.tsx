import React from 'react';
import Box from '@material-ui/core/Box';

interface TabPanelProps {
  children?: React.ReactNode;
  index: any;
  value: any;
}

function TabPanel(props: TabPanelProps) {
  const { children, value, index, ...other } = props;

  return (
    <Box hidden={value !== index} {...other} p={3}>
      {children}
    </Box>
  );
}

export default TabPanel;
