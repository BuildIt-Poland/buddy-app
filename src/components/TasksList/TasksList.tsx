import React from 'react';

const TasksList: React.FC = () => {
  return (
    <>
      <h1>Tasks List</h1>
    </>
  );
};
export default TasksList;
