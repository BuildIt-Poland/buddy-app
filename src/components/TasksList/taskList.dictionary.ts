interface TaskListDictionary {
  NO_TASKS_TITLE: string;
  NO_TASKS_SUBTITLE: string;
  TASK_STATUS_UPDATED: string;
  TASK_UPDATE_SERVER_ERROR: string;
}

const TASK_LIST_DICTIONARY: TaskListDictionary = {
  NO_TASKS_TITLE: 'No Tasks Found',
  NO_TASKS_SUBTITLE: 'You can always add a new task',
  TASK_STATUS_UPDATED: 'Task status updated',
  TASK_UPDATE_SERVER_ERROR: 'An error ocurred updating task.',
};

export default TASK_LIST_DICTIONARY;
