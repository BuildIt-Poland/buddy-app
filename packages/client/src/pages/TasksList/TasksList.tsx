import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import TabPanel from 'atoms/TabPanel';
import AvatarHeader from 'components/AvatarHeader';
import { useQuery } from '@apollo/react-hooks';
import { useSnackBar } from 'contexts/SnackbarContext';
import { QueryNewbieArgs, Query, Task } from '@buddy-app/schema';
import { TASK_LIST } from 'graphql/task-list.graphql';
import { useParams, useLocation } from 'react-router-dom';
import Box from '@material-ui/core/Box';
import TaskTabsContent from 'components/TaskTabsContent';
import { ROUTES } from 'shared/routes';
import useTaskProgress from 'hooks/useTaskProgress';
import useTaskStatusUpdate from 'hooks/useTaskStatusUpdate';
import Header, { MenuTypes, MenuColors } from 'components/Header';
import AddTaskOptions from 'components/AddTaskOptions';
import DICTIONARY from './dictionary';

const TasksList: React.FC = () => {
  const { newbieId } = useParams<QueryNewbieArgs>();
  const { state } = useLocation();
  const history = useHistory();
  const { showSnackbar } = useSnackBar();
  const defaultTabIndex = (state && state.defaultTabIndex) || 0;
  const [tabIndex, setTabIndex] = useState(defaultTabIndex);

  const { loading, data } = useQuery<Query, QueryNewbieArgs>(TASK_LIST, {
    variables: { newbieId },
  });
  const { buddyProgress } = useTaskProgress(data && data.newbie);
  const [updateTaskStatus] = useTaskStatusUpdate(newbieId, {
    onCompleted: () => showSnackbar(DICTIONARY.SUCCESS_MESSAGE),
    onError: () => showSnackbar(DICTIONARY.ERROR_MESSAGE),
  });

  const handleTabChange = (event: React.ChangeEvent<{}>, newValue: number) =>
    setTabIndex(newValue);

  const onBackClick = () => {
    history.push(ROUTES.BUDDY_SELECT_NEWBIE);
  };

  const newbieTasks = data && data.newbie.newbieTasks;
  const buddyTasks = data && data.newbie.buddyTasks;
  const pathname = ROUTES.BUDDY_ADD_TASK.replace(':newbieId', newbieId);

  return (
    <>
      <Header
        type={MenuTypes.BACK}
        color={MenuColors.PAPER}
        onButtonClick={onBackClick}>
        <AvatarHeader newbieId={newbieId} taskProgress={buddyProgress} />
        <Tabs
          centered
          value={tabIndex}
          onChange={handleTabChange}
          indicatorColor='primary'
          textColor='primary'
          variant='fullWidth'>
          <Tab label={DICTIONARY.NEWBIE_TAB_TITLE} />
          <Tab label={DICTIONARY.BUDDY_TAB_TITLE} />
        </Tabs>
      </Header>
      <Box component={'main'} flex={1} data-testid='task-list-page'>
        <TabPanel value={tabIndex} index={0}>
          <TaskTabsContent
            loading={loading}
            tabIndex={tabIndex}
            onChange={updateTaskStatus}
            tasks={newbieTasks as Task[]}
          />
        </TabPanel>
        <TabPanel value={tabIndex} index={1}>
          <TaskTabsContent
            tabIndex={tabIndex}
            loading={loading}
            onChange={updateTaskStatus}
            tasks={buddyTasks as Task[]}
          />
        </TabPanel>
        <AddTaskOptions
          title={DICTIONARY.PLUS_BUTTON_TITLE}
          to={{ pathname, state: { tabIndex } }}
        />
      </Box>
    </>
  );
};
export default TasksList;
