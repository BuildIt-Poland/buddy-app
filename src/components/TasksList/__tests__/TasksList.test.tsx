import React from 'react';
import { create } from 'react-test-renderer';

import TasksList from '../TasksList';

jest.mock('@material-ui/core/AppBar', () => 'AppBar');
jest.mock('@material-ui/core/Tabs', () => 'Tabs');
jest.mock('@material-ui/core/Tab', () => 'Tab');
jest.mock('../../TabPanel', () => 'TabPanel');

describe('Component - TasksList', () => {
  test('renders correctly', () => {
    const component = create(<TasksList />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
