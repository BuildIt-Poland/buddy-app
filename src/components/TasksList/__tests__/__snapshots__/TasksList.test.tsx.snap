// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - TasksList When there are tasks renders correctly 1`] = `
Array [
  <AppBar
    className="makeStyles-appBar-2 makeStyles-paperAppBarBackground-5 makeStyles-withoutShadow-3"
    color="inherit"
    component="header"
    position="sticky"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-1 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
      </button>
    </div>
    <AvatarHeader
      newbieId="1234"
    />
    <Tabs
      centered={true}
      indicatorColor="primary"
      onChange={[Function]}
      textColor="primary"
      value={0}
      variant="fullWidth"
    >
      <Tab
        label="Newbie Tasks"
      />
      <Tab
        label="My tasks"
      />
    </Tabs>
    <div
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-loaderShadow-6 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        style={Object {}}
      />
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        style={Object {}}
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-60 makeStyles-paperAppBarBackground-5"
    />
  </AppBar>,
  <main
    className="MuiBox-root MuiBox-root-61"
    data-testid="task-list-page"
  >
    <TabPanel
      index={0}
      value={0}
    >
      <TaskTabsContent
        loading={true}
        onChange={[Function]}
        tabIndex={0}
      />
    </TabPanel>
    <TabPanel
      index={1}
      value={0}
    >
      <TaskTabsContent
        loading={true}
        onChange={[Function]}
        tabIndex={0}
      />
    </TabPanel>
    <PlusButton
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      title="Add new task"
      to={
        Object {
          "pathname": "/buddy/newbies/1234/add-task",
          "state": Object {
            "tabIndex": 0,
          },
        }
      }
    />
  </main>,
]
`;

exports[`Component - TasksList When there are tasks renders correctly 2`] = `
Array [
  <AppBar
    className="makeStyles-appBar-2 makeStyles-paperAppBarBackground-5 makeStyles-withoutShadow-3"
    color="inherit"
    component="header"
    position="sticky"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-1 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
      </button>
    </div>
    <AvatarHeader
      newbieId="1234"
    />
    <Tabs
      centered={true}
      indicatorColor="primary"
      onChange={[Function]}
      textColor="primary"
      value={0}
      variant="fullWidth"
    >
      <Tab
        label="Newbie Tasks"
      />
      <Tab
        label="My tasks"
      />
    </Tabs>
    <div
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-loaderShadow-6 makeStyles-hideLoader-7 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        style={Object {}}
      />
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        style={Object {}}
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-72 makeStyles-paperAppBarBackground-5"
    />
  </AppBar>,
  <main
    className="MuiBox-root MuiBox-root-61"
    data-testid="task-list-page"
  >
    <TabPanel
      index={0}
      value={0}
    >
      <TaskTabsContent
        loading={false}
        onChange={[Function]}
        tabIndex={0}
        tasks={
          Array [
            Object {
              "description": "New task description",
              "id": "3",
              "status": "COMPLETED",
              "title": "New task title ",
            },
            Object {
              "description": "New task description 2",
              "id": "4",
              "status": "COMPLETED",
              "title": "New task title 2",
            },
          ]
        }
      />
    </TabPanel>
    <TabPanel
      index={1}
      value={0}
    >
      <TaskTabsContent
        loading={false}
        onChange={[Function]}
        tabIndex={0}
        tasks={
          Array [
            Object {
              "description": "New task description",
              "id": "1",
              "status": "UNCOMPLETED",
              "title": "New task title ",
            },
            Object {
              "description": "New task description 2",
              "id": "2",
              "status": "COMPLETED",
              "title": "New task title 2",
            },
          ]
        }
      />
    </TabPanel>
    <PlusButton
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      title="Add new task"
      to={
        Object {
          "pathname": "/buddy/newbies/1234/add-task",
          "state": Object {
            "tabIndex": 0,
          },
        }
      }
    />
  </main>,
]
`;

exports[`Component - TasksList when there are not tasks should render empty state 1`] = `
Array [
  <AppBar
    className="makeStyles-appBar-2 makeStyles-paperAppBarBackground-5 makeStyles-withoutShadow-3"
    color="inherit"
    component="header"
    position="sticky"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-1 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
      </button>
    </div>
    <AvatarHeader
      newbieId="1234"
    />
    <Tabs
      centered={true}
      indicatorColor="primary"
      onChange={[Function]}
      textColor="primary"
      value={0}
      variant="fullWidth"
    >
      <Tab
        label="Newbie Tasks"
      />
      <Tab
        label="My tasks"
      />
    </Tabs>
    <div
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-loaderShadow-6 makeStyles-hideLoader-7 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        style={Object {}}
      />
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        style={Object {}}
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-85 makeStyles-paperAppBarBackground-5"
    />
  </AppBar>,
  <main
    className="MuiBox-root MuiBox-root-84"
    data-testid="task-list-page"
  >
    <TabPanel
      index={0}
      value={0}
    >
      <TaskTabsContent
        loading={false}
        onChange={[Function]}
        tabIndex={0}
        tasks={Array []}
      />
    </TabPanel>
    <TabPanel
      index={1}
      value={0}
    >
      <TaskTabsContent
        loading={false}
        onChange={[Function]}
        tabIndex={0}
        tasks={Array []}
      />
    </TabPanel>
    <PlusButton
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      title="Add new task"
      to={
        Object {
          "pathname": "/buddy/newbies/1234/add-task",
          "state": Object {
            "tabIndex": 0,
          },
        }
      }
    />
  </main>,
]
`;
